<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../../img/favicon.ico">
  <title>Genomics - OARC-Rutgers</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href='https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="../../css/highlight.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Genomics";
    var mkdocs_page_input_path = "workshops/workshop.md";
    var mkdocs_page_url = "/workshops/workshop/";
  </script>
  
  <script src="../../js/jquery-2.1.1.min.js"></script>
  <script src="../../js/modernizr-2.8.3.min.js"></script>
  <script type="text/javascript" src="../../js/highlight.pack.js"></script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="../.." class="icon icon-home"> OARC-Rutgers</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
      


  
    


<ul class="" href="../..">
  
    
    
      <li class=" toctree-l1">
        <a class="" href="../..">Home</a>
      </li>
</ul>

<!--
 | This is a hacky way to pass setting from the template generator to the
 | javascript that drives the dropdown functionality
-->
<div style="width: 1px; height: 1px; visibility: none;">

</div>



  
    


<ul class="" href="../../guides/Cluster_User_Guide/">
  
    
    
      <li class=" toctree-l1">
        <a class="" href="../../guides/Cluster_User_Guide/">Cluster Guides</a>
      </li>
</ul>

<!--
 | This is a hacky way to pass setting from the template generator to the
 | javascript that drives the dropdown functionality
-->
<div style="width: 1px; height: 1px; visibility: none;">

</div>



  
    


<ul class="" href="../../guides/Cluster_Examples/">
  
    
    
      <li class=" toctree-l1">
        <a class="" href="../../guides/Cluster_Examples/">Examples</a>
      </li>
</ul>

<!--
 | This is a hacky way to pass setting from the template generator to the
 | javascript that drives the dropdown functionality
-->
<div style="width: 1px; height: 1px; visibility: none;">

</div>



  
    
    
    
    
    
    

  <p class="caption">HowTo</p>


<ul class="" href="howtos/jupyter">
  
    
      
<li class=" toctree-l1">
  <a href="../../howtos/jupyter/" title="Jupyter"
      class="">
    Jupyter
  </a>

  
  
</li>
    
      
<li class=" toctree-l1">
  <a href="../../howtos/fastx/" title="FastX"
      class="">
    FastX
  </a>

  
  
</li>
    
</ul>

<!--
 | This is a hacky way to pass setting from the template generator to the
 | javascript that drives the dropdown functionality
-->
<div style="width: 1px; height: 1px; visibility: none;">

</div>



  
    
    
    
    
    
    

  <p class="caption">Workshops</p>


<ul class="current" href="workshops/workshop">
  
    
      
<li class="current toctree-l1">
  <a href="./" title="Genomics"
      class="current">
    Genomics
  </a>

  
  
    
    
      
  
  
  
  
  <ul class="current subnav">
    
      

  
  
  


  <li class="current toctree-l2">
    
      <a href="#overview" title="Genomics"
          class="current">
        Overview
      </a>
    
    <ul class="current subnav">
      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#genomic-software" title="Genomics">Genomic Software</a>
  </li>

      
    </ul>
  </li>

    
      

  
  


  <li class=" toctree-l2">
    
      <a href="#setup" title="Genomics"
          class="">
        Setup
      </a>
    
    <ul class=" subnav">
      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#connect-to-the-cluster-login-node" title="Genomics">Connect to the cluster login node</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#get-resources-on-the-compute-node" title="Genomics">Get resources on the compute node</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#install-programs" title="Genomics">Install programs</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#download-data" title="Genomics">Download data</a>
  </li>

      
    </ul>
  </li>

    
      

  
  


  <li class=" toctree-l2">
    
      <a href="#running-bioinformatics-jobs" title="Genomics"
          class="">
        Running bioinformatics jobs
      </a>
    
    <ul class=" subnav">
      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#fastqc-raw-data-qc" title="Genomics">FastQC - raw data QC</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#trimmomatic-quality-trimadaptor-removal" title="Genomics">Trimmomatic - quality trim/adaptor removal</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#fastqc-run-on-cleaned-reads-compare-result" title="Genomics">FastQC - Run on cleaned reads, compare result</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#download-reference-and-reference-indexing" title="Genomics">Download reference and reference indexing</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#mapping-with-tophat2-star-hisat2" title="Genomics">Mapping with tophat2, (STAR, HISAT2)</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#read-counts-using-htseq-count" title="Genomics">Read counts using htseq-count</a>
  </li>

      
        

  
  


  <li class="toctree-l2 toc-item">
    <a href="#perform-mapping-qc-using-rseqc" title="Genomics">Perform Mapping QC using RSeQC</a>
  </li>

      
    </ul>
  </li>

    
  </ul>

    
  
</li>
    
      
<li class=" toctree-l1">
  <a href="../Labs/" title="Labs"
      class="">
    Labs
  </a>

  
  
</li>
    
</ul>

<!--
 | This is a hacky way to pass setting from the template generator to the
 | javascript that drives the dropdown functionality
-->
<div style="width: 1px; height: 1px; visibility: none;">

</div>



  
    


<ul class="" href="../../ressentials/">
  
    
    
      <li class=" toctree-l1">
        <a class="" href="../../ressentials/">R tutorial</a>
      </li>
</ul>

<!--
 | This is a hacky way to pass setting from the template generator to the
 | javascript that drives the dropdown functionality
-->
<div style="width: 1px; height: 1px; visibility: none;">

</div>



  
    


<ul class="" href="../../CheatSheet/">
  
    
    
      <li class=" toctree-l1">
        <a class="" href="../../CheatSheet/">CheatSheet</a>
      </li>
</ul>

<!--
 | This is a hacky way to pass setting from the template generator to the
 | javascript that drives the dropdown functionality
-->
<div style="width: 1px; height: 1px; visibility: none;">

</div>


      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../..">OARC-Rutgers</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../..">Docs</a> &raquo;</li>
    
      
        
          <li>Workshops &raquo;</li>
        
      
    
    <li>Genomics</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/rutgers-oarc/training/edit/master/docs/workshops/workshop.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="overview">Overview</h1>
<p>These are the notes from the workshop on genomics. What's covered in these notes:  <br />
1. Setup
2. Analysis
3. Labs</p>
<h2 id="genomic-software">Genomic Software</h2>
<p>This is a list of software to install for the workshop</p>
<table>
<thead>
<tr>
<th>software</th>
<th>description</th>
<th>link</th>
</tr>
</thead>
<tbody>
<tr>
<td>Seqtk</td>
<td>very handy and fast for processing fastq/a files</td>
<td><a href="https://github.com/lh3/seqtk">link</a></td>
</tr>
<tr>
<td>sratoolkit</td>
<td>downloading and processing data from GEO/SRA database</td>
<td><a href="http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.8.2/sratoolkit.2.8.2-centos_linux64.tar.gz">link</a></td>
</tr>
<tr>
<td>htseq-count</td>
<td>counting the  reads mapped on to genomics feature</td>
<td><a href="http://htseq.readthedocs.io/en/master/install.html">link</a></td>
</tr>
<tr>
<td>fastQC</td>
<td>widely used for sequencing read QC</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.6.zip">link</a></td>
</tr>
<tr>
<td>RSeQC-2.6.4</td>
<td>An RNA-seq quality control package, multiple functions</td>
<td>python package</td>
</tr>
<tr>
<td>trimmomatic</td>
<td>fastq quality trim and adaptor removal</td>
<td><a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.36.zip">link</a></td>
</tr>
</tbody>
</table>
<p>This is a list of software already available on the cluster and the command you need to execute to load it in your environment: </p>
<table>
<thead>
<tr>
<th>software</th>
<th>description</th>
<th>load it on the cluster</th>
</tr>
</thead>
<tbody>
<tr>
<td>Samtools</td>
<td></td>
<td><code>module load samtools</code></td>
</tr>
<tr>
<td>Bedtools</td>
<td></td>
<td><code>module load bedtools2./2.25.0</code></td>
</tr>
<tr>
<td>bowtie2</td>
<td>alignment software</td>
<td><code>module load bowtie2</code></td>
</tr>
<tr>
<td>tophat2</td>
<td>alignment software</td>
<td><code>module load mvapich2/2.1  boost/1.59.0  tophat2/2.1.0</code></td>
</tr>
<tr>
<td>R</td>
<td>language for statistical analysis</td>
<td><code>module load intel/17.0.4 R-Project/3.4.1</code></td>
</tr>
</tbody>
</table>
<p>This is a list of other software you might find useful:</p>
<table>
<thead>
<tr>
<th>software</th>
<th>description</th>
<th>link</th>
</tr>
</thead>
<tbody>
<tr>
<td>GSEA</td>
<td>genome set enrichment analysis</td>
<td><a href="http://software.broadinstitute.org/gsea/index.jsp">link</a></td>
</tr>
<tr>
<td>IGV</td>
<td>Interactive Genome Viewer</td>
<td><a href="http://software.broadinstitute.org/software/igv/">link</a></td>
</tr>
<tr>
<td>Cytoscape</td>
<td>Network visualization softwar</td>
<td><a href="http://www.cytoscape.org/">link</a></td>
</tr>
</tbody>
</table>
<h1 id="setup">Setup</h1>
<h2 id="connect-to-the-cluster-login-node">Connect to the cluster login node</h2>
<p>The preferred method to connect to the cluster is through a web browser and fastX client</p>
<ul>
<li>
<p><strong>via FastX</strong>: in your browser, go to <a href="https://amarel.hpc.rutgers.edu:3443">this website</a> . (It will only work from campus or if you are connected via VPN.) See <a href="../../howtos/fastx">Fastx</a> for a walkthrough that includes a tip on pasting commands. </p>
</li>
<li>
<p><strong>via a terminal</strong>: if you have a Mac or Linux, terminal is part of your standard apps. If you have Windows, install an SSH client such as  [mobaXterm] (https://mobaxterm.mobatek.net/). Then from your terminal connect to the cluster by executing the following command: <br />
<code>ssh -X &lt;your net id&gt;@amarel.hpc.rutgers.edu</code>   </p>
</li>
</ul>
<h2 id="get-resources-on-the-compute-node">Get resources on the compute node</h2>
<p>You get to the cluster to execute your computations by running the following command in your terminal:  <br />
<code>srun  -p main --reservation=genomics -N 1 -c 2 -n 1 -t 01:40:00 --export=ALL --pty /bin/bash</code> <br />
Notice that the name in your terminal will change from <code>amarel</code> to node name like <code>hal0025</code> or <code>slepner086</code>. This means that you will not impede other users who are also using the login node, and will be placed on a machine which you share with only a few people. The following table explains the parts of this command: </p>
<table>
<thead>
<tr>
<th>command part</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>srun</code></td>
<td><code>slurm</code> run, i.e. allocate resources and run via <code>slurm</code> scheduler</td>
</tr>
<tr>
<td><code>-p main</code></td>
<td>on the main partition, one of several queues on the cluster</td>
</tr>
<tr>
<td><code>--reservation=genomics</code></td>
<td>we reserved some compute nodes for this workshop to not wait long for resources</td>
</tr>
<tr>
<td><code>-N 1</code></td>
<td>ask for one node</td>
</tr>
<tr>
<td><code>-c 2</code></td>
<td>ask for two cores</td>
</tr>
<tr>
<td><code>-n 1</code></td>
<td>this will be 1 most times</td>
</tr>
<tr>
<td><code>-t 01:40:00</code></td>
<td>run this for a maximum time of 1 hour 40 minutes</td>
</tr>
<tr>
<td><code>--pty /bin/bash</code></td>
<td>run the terminal shell in an interactive mode</td>
</tr>
</tbody>
</table>
<h3 id="prepare-some-directories-for-the-data">Prepare some directories for the data</h3>
<p>You have two main spaces on the Amarel cluster. These are: </p>
<ul>
<li>your home directory - <code>/home/netid/</code>  (e.g. <code>/home/kp807/</code> for my netid) -</li>
<li>your scratch directory - <code>/scratch/netid/</code></li>
</ul>
<p>They differ in how often they are backed up and by size (100Gb for <code>/home</code> and 500Gb for <code>/scratch</code>). So we will install programs in <code>/home</code>, while the data and output will be in <code>/scratch</code>. Execute these commands: </p>
<pre><code>                cd ~                      # change directory to your home directory
                mkdir Genomics_Workshop
                cd Genomics_Workshop
                mkdir Programs            # download and install programs here
</code></pre>

<pre><code>                mkdir -p  /scratch/$USER/Genomics_Workshop/
                cd /scratch/$USER/Genomics_Workshop/
                mkdir untreated  
                mkdir dex_treated
</code></pre>

<h2 id="install-programs">Install programs</h2>
<p>Each program will have slightly different installation instructions. Here is a handy sequence of commands that will install them: </p>
<pre><code>##We are going to do some modifications to a system file .bashrc, be careful doing it and make sure that you created a copy of your .bashrc file
                cd
                cp .bashrc .bashrc_20180118
                nano .bashrc
##At the end of the file add the line  “##  Genomics_Workshop 06/27/2018 settings”
##Exit nano (ctrl+x)

#Seqtk:   https://github.com/lh3/seqtk   ##  very handy and fast for processing fastq/a files
                cd
                cd Genomics_Workshop/Programs/
                git clone https://github.com/lh3/seqtk.git 
                cd seqtk
                make
                echo ‘export PATH=$HOME/Genomics_Workshop/Programs/seqtk:$PATH’ &gt;&gt;  ~/.bashrc
                source ~/.bashrc

#sratoolkit     https://www.ncbi.nlm.nih.gov/books/NBK158900/
                        https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software
##for downloading and processing data from GEO/SRA database
                cd
                cd Genomics_Workshop/Programs/
                wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.8.2/sratoolkit.2.8.2-centos_linux64.tar.gz
                tar zxvf sratoolkit.2.8.2-centos_linux64.tar.gz
                echo ‘export PATH=$HOME/Genomics_Workshop/Programs/sratoolkit.2.8.2-centos_linux64/bin:$PATH’ &gt;&gt; ~/.bashrc
                source ~/.bashrc 

#htseq-count    http://htseq.readthedocs.io/en/master/install.html ##for counting the  reads mapped on to genomics feature

                cd
                module load intel/17.0.2 python/2.7.12
                pip install HTSeq --user

#fastQC          #widely used for sequencing read QC

                cd
                cd  Genomics_Workshop/Programs/
                wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.6.zip
                unzip  fastqc_v0.11.6.zip
                echo ‘export PATH=$HOME/Genomics_Workshop/Programs/FastQC:$PATH’ &gt;&gt; ~/.bashrc
                source ~/.bashrc


#FASTX-toolkit    http://hannonlab.cshl.edu/fastx_toolkit/   (##also a tool kit for fastq processing, quality trim, adaptor removal, etc. try if time allows)

#RSeQC-2.6.4     ##An RNA-seq quality control package, multiple functions

                cd
                cd  Genomics_Workshop/Programs/
                module load python/2.7.12
                module load intel/17.0.2
                pip install RSeQC --user


#trimmomatic             ##for fastq quality trim and adaptor removal

                cd
                cd  Genomics_Workshop/Programs/
                wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.36.zip
                unzip Trimmomatic-0.36.zip 
                java -jar ~/Genomics_Workshop/Programs/Trimmomatic-0.36/trimmomatic-0.36.jar #-h                    

</code></pre>

<p>Moreover, execute the following commands to load system-installed sofware so the system knows where to find it (i.e. <code>samtools</code> command will work if you execute <code>module load samtools</code>): </p>
<pre><code>module load samtools       
module load bedtools2./2.25.0
module load bowtie2
module load mvapich2/2.1  boost/1.59.0  tophat2/2.1.0
module load intel/17.0.4 R-Project/3.4.1
</code></pre>

<h2 id="download-data">Download data</h2>
<p>We will download human RNA-seq data with <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52778">GEO accession GSE52778</a>. The samples we download are in NCBI's short read archive format (SRA). To unpack the original sequence files can be a bit tricky at first. Please put them in different directories:</p>
<pre><code>                mkdir -p  /scratch/$USER/Genomics_Workshop/
                cd /scratch/$USER/Genomics_Workshop/
                mkdir untreated  
                mkdir dex_treated
</code></pre>

<p>We will use sratoolkit programs to download data but first we need to configure a location where all data files will be stored. <code>sratoolkit</code> will be in your home directory, under <code>Programs</code>, and the <code>vdb-config</code> might be under the <code>bin</code> directory. You will enter <code>/scratch/your_netID/Genomics_Workshop/download</code> for the path - NOTE you have to replace <code>your_netID</code> with your true netId, e.g. <code>kp807</code>. Do not copy blindly! So your downloads will always go to this directory and you will need to move it out to wherever you want to have them. </p>
<pre><code>                vdb-config   --interactive-mode textual     ### dash-dash before interactive-mode
                         Your choice &gt; 4
## type new path in
                        /scratch/your_netID/Genomics_Workshop/download
                        Your choice &gt; Y
</code></pre>

<p>Then execute the following commands to get the data. Both <code>prefetch</code> and <code>fastq-dump</code> are part of sratools. Downloading can take some time! [TODO: check how much time for these files!]</p>
<pre><code>prefetch -v SRR1039508                           # fetches the SRA data
fastq-dump --gzip --split-files SRR1039508       # ???? 
</code></pre>

<p>You have to pay attention to where you are putting your data. So these two commands will actually be several: </p>
<pre><code>                cd  untreated                       # now you are in /scratch/..../Genomics_Workshop/untreated
                prefetch -v SRR1039508
                mv /scratch/$USER/Genomics_Workshop/download/sra/SRR1039508.sra .  # moving from download to actual directory 
                fastq-dump --gzip --split-files SRR1039516
</code></pre>

<p>The commands above showed how to do it for one sample. You need to do it for 6 samples total. </p>
<pre><code>                SRR1039508  SRR1039512 SRR1039516   (untreated)
                SRR1039509  SRR1039513  SRR1039517  (dex_treated)
</code></pre>

<h1 id="running-bioinformatics-jobs">Running bioinformatics jobs</h1>
<h2 id="fastqc-raw-data-qc">FastQC - raw data QC</h2>
<p>Explain what is fastqc is doing here - TODO</p>
<pre><code>        cd /scratch/$USER/Genomics_Workshop/untreated         
        module load java  ## fastqc is written in java; we need to load java before using fastqc
        mkdir fastqc      ## create a folder to store the QC output 
        fastqc -o fastqc SRR1039508_1.fastq SRR1039508_2.fastq
</code></pre>

<p>FastQC produces an html page as output, <code>fastqc/SRR1039508_1_fastqc.html</code>, with different kinds of views of data (and Phred scores). You can download this file to your local machine and open it in browser. It is also possible to open browser on the cluster, but the cluster is not really designed for that. To see more about FastQC, see this pdf file - /projects/oarc/Genomics_Workshop/Labs/FastQC_details.pdf</p>
<h2 id="trimmomatic-quality-trimadaptor-removal">Trimmomatic - quality trim/adaptor removal</h2>
<pre><code>    ##for demonstration purpose, we will take a small subset data using seqtk
    cd /scratch/$USER/Genomics_Workshop/untreated
    seqtk sample -s100  SRR1039508_1.fastq 10000 &gt; SRR1039508_1_10k.fastq 
    seqtk sample -s100  SRR1039508_2.fastq 10000 &gt; SRR1039508_2_10k.fastq 
    ## /projects/oarc/Genomics_Workshop/Labs/Seqtk_Examples.docx
    ## This file contains useful examples how to use seqtk

    ##now, run trimmomatic to trim the read quality , and remove adaptor
    module load java    ### because trimmomatic
    java -jar /home/$USER/Genomics_Workshop/Programs/Trimmomatic-0.36/trimmomatic-0.36.jar PE -phred33 -trimlog trim.log SRR1039508_1_10k.fastq SRR1039508_2_10k.fastq SRR1039508_1.paired.fastq SRR1039508_1.unpaired.fastq SRR1039508_2.paired.fastq SRR1039508_2.unpaired.fastq ILLUMINACLIP:/home/$USER/Programs/Trimmomatic-0.36/adapters/TruSeq3-PE.fa:2:30:10 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:35
</code></pre>
<p><strong>NOTE:</strong>  the above is a one line command, illustrated as the following:</p>
<pre><code>        java -jar trimmomatic-0.36.jar PE \
        -phred33 -trimlog trim.log \
        input_1.fq  input_2.fq \
        output_1_paired.fq  output_1_unpaired.fq \
        output_2_paired.fq  output_2_unpaired.fq \
        ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:35 / 

        ## Once it started run, you shall see the following:
        TrimmomaticPE: Started with arguments:
        -phred33 -trimlog trim.log SRR1039508_1_10k.fastq SRR1039508_2_10k.fastq SRR1039508_1.paired.fastq SRR1039508_1.unpaired.fastq SRR1039508_2.paired.fastq SRR1039508_2.unpaired.fastq ILLUMINACLIP:/home/yc759/Programs/Trimmomatic-0.36/adapters/TruSeq3-PE.fa:2:30:10 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:35
        Multiple cores found: Using 2 threads
        Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
        ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
        Input Read Pairs: 100000 Both Surviving: 96596 (96.60%) Forward Only Surviving: 1542 (1.54%) Reverse Only Surviving: 1467 (1.47%) Dropped: 395 (0.40%)
        TrimmomaticPE: Completed successfully

        ##view the output, the trim.log file, .e.g.  length=63 55 1 56 7 (the original read length 63, now 55 after trim, 1 base from left end and 7 bases from the right end were trimmed off, 56 bases in middle remained)

        ##you may also try fastx_quality_stats from the FASTX—toolkit

</code></pre>

<h2 id="fastqc-run-on-cleaned-reads-compare-result">FastQC - Run on cleaned reads, compare result</h2>
<pre><code>     module load java
     fastqc -o fastqc SRR1039508_1.paired.fastq SRR1039508_2.paired.fastq

     ## /projects/oarc/Genomics_Workshop/Labs/FastQC_details.pdf , helpful in viewing and interpreting the output
</code></pre>

<h2 id="download-reference-and-reference-indexing">Download reference and reference indexing</h2>
<p>Human genome indexing will take hours, we have the reference pre-prepared. Stored at  <code>/projects/oarc/Genomics_Workshop/Reference/</code>
For in class practice, we will do this on E.coli genome</p>
<pre><code>        cd /scratch/$USER/Genomics_Workshop/
        mkdir Reference
        cd Reference

        wget ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/Escherichia_coli/latest_assembly_versions/GCA_000005845.2_ASM584v2/GCA_000005845.2_ASM584v2_genomic.fna.gz

        gunzip GCA_000005845.2_ASM584v2_genomic.fna.gz
        module load bowtie2
        bowtie2-build GCA_000005845.2_ASM584v2_genomic.fna GCA_000005845.2_ASM584v2_genomic

        ##if download from ENSEMBLE
        wget ftp://ftp.ensemblgenomes.org/pub/bacteria/release-38/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655/dna/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.dna.toplevel.fa.gz
</code></pre>

<h2 id="mapping-with-tophat2-star-hisat2">Mapping with tophat2, (STAR, HISAT2)</h2>
<p>Now, go to your data folder
        <code>cd  /scratch/$USER/Genomics_Workshop/untreated</code></p>
<pre><code>        cd  /scratch/$USER/Genomics_Workshop/untreated
        module load mvapich2/2.1  boost/1.59.0  tophat2/2.1.0
        module load samtools   #bowtie2 is loaded already
        mkdir tophat_out
        tophat2 -p 10 --library-type fr-unstranded  -o tophat_out/untreated_SRR1039508_1
        0k --transcriptome-index /projects/oarc/Genomics_Workshop/Reference/  hg20_transciptome/GR
        Ch38.78 /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.dna.toplevel
        SRR1039508_1.paired.fastq SRR1039508_2.paired.fastq
        ## you shall modify the -p value to be consistent with the -c value you requested in the beginning
</code></pre>

<p>You shall see something like:</p>
<pre><code>   [2018-03-30 11:48:57] Beginning TopHat run (v2.1.0)
-----------------------------------------------
[2018-03-30 11:48:57] Checking for Bowtie
                  Bowtie version:        2.2.9.0
[2018-03-30 11:48:58] Checking for Bowtie index files (transcriptome)..
[2018-03-30 11:48:58] Checking for Bowtie index files (genome)..
[2018-03-30 11:48:58] Checking for reference FASTA file
[2018-03-30 11:48:58] Generating SAM header for /projects/oarc/Genomics_Workshop/Referen
ce/hg20/Homo_sapiens.GRCh38.dna.toplevel
[2018-03-30 11:49:09] Reading known junctions from GTF file
[2018-03-30 11:49:27] Preparing reads
         left reads: min. length=35, max. length=63, 96592 kept reads (4 discarded)
        right reads: min. length=35, max. length=63, 96594 kept reads (2 discarded)
[2018-03-30 11:49:29] Using pre-built transcriptome data..
[2018-03-30 11:49:35] Mapping left_kept_reads to transcriptome GRCh38.78 with Bowtie2
[2018-03-30 11:49:49] Mapping right_kept_reads to transcriptome GRCh38.78 with Bowtie2
[2018-03-30 11:50:03] Resuming TopHat pipeline with unmapped reads
[2018-03-30 11:50:03] Mapping left_kept_reads.m2g_um to genome Homo_sapiens.GRCh38.dna.t
oplevel with Bowtie2
[2018-03-30 11:50:16] Mapping left_kept_reads.m2g_um_seg1 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (1/2)
[2018-03-30 11:50:18] Mapping left_kept_reads.m2g_um_seg2 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (2/2)
[2018-03-30 11:50:20] Mapping right_kept_reads.m2g_um to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2
[2018-03-30 11:50:23] Mapping right_kept_reads.m2g_um_seg1 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (1/2)
[2018-03-30 11:50:25] Mapping right_kept_reads.m2g_um_seg2 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (2/2)
……………………………………………………….
</code></pre>

<p>The transcriptome index was built  by pointing to gtf file first,  here we have it prepared already, just so we can save time.  The following would be the command to generate the transcriptome index while running tophat alignment.</p>
<pre><code>   tophat2 -p 10 --library-type fr-unstranded  -o tophat_out/untreated_SRR1039516 –GTF /projects/oarc/Genomics_Workshop/Reference/hg20/ Homo_sapiens.GRCh38.78.gtf --transcriptome-index /projects/oarc/Genomics_Workshop/Reference/hg20_transciptome/GRCh38.78 /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.dna.toplevel SRR1039516_1.fastq.pairedOut.fastq SRR1039516_2.fastq.pairedOut.fastq

</code></pre>

<p>The output folder <code>tophat_out/untreated_SRR1039508/</code> shall contain the following files/folder (in blue):<br />
<code>cd  /projects/oarc/Genomics_Workshop/SRA_data/untreated/tophat_out/untreated_SRR1039508</code>
List the contents of this directory: </p>
<pre><code>$ ll
   total 2183632
 -rw-rw-r-- 1 yc759 oarc 2174796848 Jan 16 21:57 accepted_hits.bam
 -rw-rw-r-- 1 yc759 oarc        565 Jan 16 21:57 align_summary.txt
 -rw-rw-r-- 1 yc759 oarc    1921529 Jan 16 21:57 deletions.bed
 -rw-rw-r-- 1 yc759 oarc    2239884 Jan 16 21:57 insertions.bed
 -rw-rw-r-- 1 yc759 oarc   14181618 Jan 16 21:57 junctions.bed
 drwxrwsr-x 2 yc759 oarc       4096 Jan 16 21:57 logs
 -rw-rw-r-- 1 yc759 oarc        184 Jan 16 21:57 prep_reads.info
 -rw-rw-r-- 1 yc759 oarc   42846571 Jan 16 21:57 unmapped.bam
</code></pre>

<h2 id="read-counts-using-htseq-count">Read counts using htseq-count</h2>
<p>Remember where your alignment output folder is. Previously it was <code>/scratch/$USER/Genomics_Workshop/untreated/tophat_out/untreated_SRR1039508</code>. If you managed to produce the bam file yourself great. If not, let's link the one we prepared for you. We use <code>ln -s</code> command to do "soft link" which makes it possible to refer to the data without physically copying it. </p>
<pre><code>     cd /scratch/$USER/Genomics_Workshop/untreated/tophat_out/untreated_SRR1039508 
     ln -s /projects/oarc/Genomics_Workshop/SRA_data/untreated/tophat_out/untreated_SRR1039508/accepted_hits.bam accepted_hits.bam   
</code></pre>

<p>Next, we will use samtools to sort the bam file by name:  because htseq-count accepts bam file sorted by <strong>name</strong> as default, but tophat generates bam sorted by <strong>coordinates</strong> by default. </p>
<pre><code>     module load samtools intel/17.0.2 python/2.7.12   #loads the software
     samtools sort -n  accepted_hits.bam | samtools view | htseq-count -m intersection-nonempty -t exon -i gene_id -s no --additional-attr=gene_name  -/projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.78.gtf &gt; untreated08.txt
</code></pre>

<p>In the same way, generate the counts file <code>untreated12.txt</code>, <code>untreated16.txt</code>,<code>dex09.txt</code>, <code>dex13.txt</code>, <code>dex17.txt</code>. </p>
<h2 id="perform-mapping-qc-using-rseqc">Perform Mapping QC using RSeQC</h2>
<p>Now,  quality control using RSeQC - a few examples are given here, please go to the <a href="http://rseqc.sourceforge.net/">rseqc website</a> for more functions. Execute:   </p>
<pre><code>cd /scratch/$USER/Genomics_Workshop/untreated/tophat_out/untreated_SRR1039508 
module load python/2.7.12
module load intel/17.0.4

read_distribution.py -i accepted_hits.bam -r /projects/oarc/Genomics_Workshop/Reference/Homo_sapiens.GRCh38.79.bed
</code></pre>

<p>Output: </p>
<pre><code>        processing/projects/oarc/Genomics_Workshop/Reference/Homo_sapiens.GRCh38.79.bed ... Done
        processing accepted_hits.bam ... Finished

        Total Reads                   43474036
        Total Tags                    54438789
        Total Assigned Tags           53991382
        =====================================================================
        Group               Total_bases         Tag_count           Tags/Kb
        CDS_Exons           103371993           43264842            418.54
        5'UTR_Exons         5217678             583447              111.82
        3'UTR_Exons         29324747            8145122             277.76
        Introns             1500197093          1805034             1.20
        TSS_up_1kb          33306654            18893               0.57
        TSS_up_5kb          148463534           41165               0.28
        TSS_up_10kb         265823549           55644               0.21
        TES_down_1kb        35215293            50954               1.45
        TES_down_5kb        152556214           113325              0.74
        TES_down_10kb       268614580           137293              0.51
        =====================================================================
</code></pre>

<p>Execute:<br />
<code>bam_stat.py -i accepted_hits.bam</code>
Output: </p>
<pre><code>        Load BAM file ...

        Done

        #==================================================
        #All numbers are READ count
        #==================================================

        Total records:                          52528699

        QC failed:                              0
        Optical/PCR duplicate:                  0
        Non primary hits                        9054663
        Unmapped reads:                         0
        mapq &lt; mapq_cut (non-unique):           2684801

        mapq &gt;= mapq_cut (unique):              40789235
        Read-1:                                 20414530
        Read-2:                                 20374705
        Reads map to '+':                       20393901
        Reads map to '-':                       20395334
        Non-splice reads:                       30860931
        Splice reads:                           9928304
        Reads mapped in proper pairs:           32386536
        Proper-paired reads map to different chrom:312

</code></pre>

<p>The script does genebody coverage calculation requires the input bam files to be sorted and indexed (we will do it using samtools). The calculation and plot will require R
Go to one of the tophat_out sample folder</p>
<pre><code>module load intel/17.0.4  R-Project/3.4.1    
module load samtools
samtools sort accepted_hits.bam –o accepted_hits.sorted.bam 
##this may take a while, you may use the one already prepared for you by making a soft link

ln -s /projects/oarc/Genomics_Workshop/SRA_data/untreated/tophat_out/untreated_SRR1039508/accepted_hits.sorted.bam accepted_hits.sorted.bam


samtools index accepted_hits.sorted.bam

geneBody_coverage.py -r /projects/oarc/Genomics_Workshop/Reference/hg38.housekeepingGenes.bed -i accepted_hits.sorted.bam -o test
@ 2018-01-14 13:17:33: Read BED file (reference gene model) ...
@ 2018-01-14 13:17:33: Total 3802 transcripts loaded
@ 2018-01-14 13:17:33: Get BAM file(s) ...
        accepted_hits.sorted.bam
@ 2018-01-14 13:17:33: Processing accepted_hits.sorted.bam ...
        3800 transcripts finished


        Sample  Skewness
        accepted_hits.sorted    -3.61577607436
@ 2018-01-14 13:28:59: Running R script ...
null device
          1

</code></pre>

<p>output files:        test.geneBodyCoverage.r
                                test.geneBodyCoverage.txt
                log.txt
                test.geneBodyCoverage.curves.pdf</p>
<p>download the gene.bed files :  <code>https://sourceforge.net/projects/rseqc/files/BED/Human_Homo_sapiens/</code><br />
Be careful that the genome version, be consistent between reference genome used in mapping and now. For now, you may use what’s provided </p>
<p>if you want to find out whether the sequencing read is strand specific or not, do:</p>
<pre><code>        $ infer_experiment.py -r  /projects/oarc/Genomics_Workshop/Reference/ Homo_sapiens.GRCh38.79.bed -i accepted_hits.bam

    Reading reference gene model /projects/oarc/Genomics_Workshop/Reference/Homo_sapiens.GRCh38.79.bed ... Done
    Loading SAM/BAM file ...  Total 200000 usable reads were sampled


This is PairEnd Data
Fraction of reads failed to determine: 0.1406
Fraction of reads explained by &quot;1++,1--,2+-,2-+&quot;: 0.4302
Fraction of reads explained by &quot;1+-,1-+,2++,2--&quot;: 0.4292
</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../Labs/" class="btn btn-neutral float-right" title="Labs">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../../howtos/fastx/" class="btn btn-neutral" title="FastX"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/rutgers-oarc/training/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../../howtos/fastx/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../Labs/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme.js"></script>
      <script src="../../search/require.js"></script>
      <script src="../../search/search.js"></script>

</body>
</html>
