<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../../img/favicon.ico">
  <title>Overview - OARC-Rutgers</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href='https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Overview";
    var mkdocs_page_input_path = "workshops/genomics_1.md";
    var mkdocs_page_url = "/training/workshops/genomics_1/";
  </script>
  
  <script src="../../js/jquery-2.1.1.min.js" defer></script>
  <script src="../../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="../.." class="icon icon-home"> OARC-Rutgers</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="tocbase current">
    
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../..">Home</a>
  </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../guides/Cluster_User_Guide/">Cluster Guides</a>
  </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../guides/Cluster_Examples/">Examples</a>
  </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../guides/community_resources/">Community</a>
  </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../guides/classes/">Classes</a>
  </li>
    
      
  <li class="navtree toctree-l1 label">
    <p class="caption">HowTo</p>
  </li>


  

  
    <li class="navtree toctree-l1 group">
      <ul class="navtree subnav-l1 current">
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../../howtos/jupyter/">Jupyter</a>
  </li>
        
          


  <li class="navtree toctree-l2 inactive">
    <a class="" href="../../howtos/fastx/">FastX</a>
  </li>
        
      </ul>
    </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../ressentials/">R tutorial</a>
  </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../CheatSheet/">Cheat Sheets</a>
  </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../resources/">Resources</a>
  </li>
    
      


  <li class="navtree toctree-l1 inactive">
    <a class="" href="../../faq/">FAQ</a>
  </li>
    
  </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../..">OARC-Rutgers</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../..">Docs</a> &raquo;</li>
    
      
    
    <li>Overview</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/rutgers-oarc/training/edit/master/docs/workshops/genomics_1.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="overview">Overview</h1>
<h2 id="genomic-software">Genomic Software</h2>
<p>This is a list of software to install for the workshop</p>
<table>
<thead>
<tr>
<th>Software</th>
<th>Description</th>
<th>URL</th>
</tr>
</thead>
<tbody>
<tr>
<td>seqtk</td>
<td>very handy and fast for processing fastq/a files</td>
<td><a href="https://github.com/lh3/seqtk">[link]</a></td>
</tr>
<tr>
<td>sratoolkit</td>
<td>downloading and processing data from GEO/SRA database</td>
<td><a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft/">[link]</a> <a href="http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.8.2/sratoolkit.2.8.2-centos_linux64.tar.gz">[download]</a></td>
</tr>
<tr>
<td>htseq-count</td>
<td>counting the  reads mapped on to genomics feature</td>
<td><a href="http://htseq.readthedocs.io/en/master/install.html">[link]</a></td>
</tr>
<tr>
<td>fastQC</td>
<td>widely used for sequencing read QC</td>
<td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc">[link]</a> <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.6.zip">[download]</a></td>
</tr>
<tr>
<td>RSeQC-2.6.4</td>
<td>An RNA-seq quality control package, multiple functions</td>
<td><a href="http://rseqc.sourceforge.net/">[link]</a></td>
</tr>
<tr>
<td>trimmomatic</td>
<td>fastq quality trim and adaptor removal</td>
<td><a href="http://www.usadellab.org/cms/?page=trimmomatic">[link]</a> <a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.36.zip">[download]</a></td>
</tr>
</tbody>
</table>
<p>This is a list of software already available on the cluster and the command you need to execute to load it in your environment: </p>
<table>
<thead>
<tr>
<th>Software</th>
<th>Description</th>
<th>Access on the cluster</th>
</tr>
</thead>
<tbody>
<tr>
<td>samtools</td>
<td>utilities for short DNA seq alignments <a href="http://samtools.sourceforge.net/">[link]</a></td>
<td><code>module load samtools/1.3.1</code></td>
</tr>
<tr>
<td>bedtools</td>
<td>tools for a wide-range of genomics analysis <a href="http://bedtools.readthedocs.io/en/latest/">[link]</a></td>
<td><code>module load bedtools2/2.25.0</code></td>
</tr>
<tr>
<td>bowtie2</td>
<td>alignment software <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">[link]</a></td>
<td><code>module load bowtie2/2.2.9</code></td>
</tr>
<tr>
<td>tophat2</td>
<td>a fast splice junction mapper for RNA-Seq reads <a href="https://ccb.jhu.edu/software/tophat/index.shtml">[link]</a></td>
<td><code>module load mvapich2/2.1  boost/1.59.0 tophat2/2.1.0</code></td>
</tr>
<tr>
<td>R</td>
<td>language for statistical analysis</td>
<td><code>module load intel/17.0.4 R-Project/3.4.1</code></td>
</tr>
</tbody>
</table>
<p>This is a list of other software you might find useful:</p>
<table>
<thead>
<tr>
<th>Software</th>
<th>Description</th>
<th>URL</th>
</tr>
</thead>
<tbody>
<tr>
<td>GSEA</td>
<td>Genome set enrichment analysis</td>
<td><a href="http://software.broadinstitute.org/gsea/index.jsp">[link]</a></td>
</tr>
<tr>
<td>IGV</td>
<td>Interactive Genome Viewer</td>
<td><a href="http://software.broadinstitute.org/software/igv/">[link]</a></td>
</tr>
<tr>
<td>Cytoscape</td>
<td>Network visualization softwar</td>
<td><a href="http://www.cytoscape.org/">[link]</a></td>
</tr>
</tbody>
</table>
<h1 id="setup">Setup</h1>
<h2 id="connect-to-the-cluster-login-node">Connect to the cluster login node</h2>
<p>The preferred method to connect to the cluster is through a web browser and fastX client</p>
<ul>
<li><strong>via FastX</strong>: in your browser, go to https://amarel.hpc.rutgers.edu:3443 or <a href="https://amarel.hpc.rutgers.edu:3443">click here</a>. (It will only work from campus or if you are connected via VPN <a href="https://ssl-vpn.rutgers.edu/dana-na/auth/url_default/welcome.cgi">link</a>.) See <a href="https://rutgers-oarc.github.io/training/howtos/fastx/">FastX</a> for a walkthrough that includes a tip on copying/pasting commands. </li>
</ul>
<table>
<thead>
<tr>
<th>Login page</th>
<th>Click on Launch Session and Select XFCE desktop</th>
<th>Open new terminal, copy text from your computer into clipboard window and click on Send to Remote. Put cursor into a terminal and click Edit-&gt;Paste. The text from a clipboard appears in the terminal. Hit Return to submit a job to a compute node. Notice that the jobid number has been assigned and resources have been allocated. The command prompt in the treminal will also change from a login node netid@amarel to a compute node netid@slepner or netid@hal.</th>
</tr>
</thead>
<tbody>
<tr>
<td><img alt="" src="https://github.com/rutgers-oarc/training/blob/master/workshops/fastX_web.png" title="FastX web interface" /></td>
<td><img alt="" src="https://github.com/rutgers-oarc/training/blob/master/workshops/FX_newsession.png" title="FastX new session" /></td>
<td><img alt="" src="https://github.com/rutgers-oarc/training/blob/master/workshops/FX_srun.png" title="FastX terminal srun" /></td>
</tr>
</tbody>
</table>
<p>On rare occasions, especially if the user has a modified .bashrc file,  FastX doesn't work. If it is the case, you can still use a terminal but the connection will be much slower for graphical outputs.
  - <strong>via a terminal</strong>: if you have a Mac or Linux, a terminal is part of your standard apps. If you have Windows, install an SSH client such as  <a href="https://mobaxterm.mobatek.net/">mobaXterm</a>. Then from your terminal connect to the cluster by executing the following command: <br />
<code>ssh -X &lt;your netid&gt;@amarel.hpc.rutgers.edu</code>   </p>
<h2 id="do-not-run-any-computational-jobs-on-the-login-node">DO NOT RUN ANY COMPUTATIONAL JOBS ON THE LOGIN NODE</h2>
<h2 id="get-resources-on-the-compute-node">Get resources on the compute node</h2>
<p>When you login to the cluster you are on the login node. Jobs are not allowed to be run on the loging node, intstead you need to request a resource on the compute node for your job. This means that you will not impede other users who are also using the login node, and will be placed on a machine which you share with only a few people. You can do so by running the following command in your terminal:    </p>
<pre><code>srun  -p main --x11 --reservation=genomics -N 1 -c 2 -n 1 -t 01:40:00 --pty /bin/bash -i
</code></pre>

<p>or just run this script</p>
<pre><code>node_request.sh
</code></pre>

<p>Notice that the name in your terminal will change from <code>amarel</code> to a node name like <code>hal0025</code> or <code>slepner086</code>.  The following table explains the parts of this command: </p>
<table>
<thead>
<tr>
<th>command part</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>srun</code></td>
<td><code>slurm</code> run, i.e. allocate resources and run via <code>slurm</code> scheduler</td>
</tr>
<tr>
<td><code>-p main</code></td>
<td>on the main partition, one of several queues on the cluster</td>
</tr>
<tr>
<td><code>--x11</code></td>
<td>it allows the graphical output from a compute node, e.g. GUI of the program</td>
</tr>
<tr>
<td><code>--reservation=genomics</code></td>
<td>we reserved some compute nodes for this workshop to avoid waiting in the queue</td>
</tr>
<tr>
<td><code>-N 1</code></td>
<td>ask for one compute node</td>
</tr>
<tr>
<td><code>-c 2</code></td>
<td>ask for two cpu cores</td>
</tr>
<tr>
<td><code>-n 1</code></td>
<td>tells slurm that the job will be run as  1 task ( for parallel MPI jobs it could be more than 1</td>
</tr>
<tr>
<td><code>-t 01:40:00</code></td>
<td>run this for a maximum time of 1 hour 40 minutes</td>
</tr>
<tr>
<td><code>--pty /bin/bash -i</code></td>
<td>run the terminal shell in an interactive mode</td>
</tr>
</tbody>
</table>
<h3 id="understanding-your-data-space">Understanding your data space</h3>
<p>You have two main spaces on the Amarel cluster. These are: </p>
<ul>
<li>your home directory (100Gb) - <code>/home/netid/</code> </li>
<li>your scratch directory (500Gb)- <code>/scratch/netid/</code> </li>
</ul>
<p>They differ in how they are backed up (scratch is not backed up) and by read/write speed. <br> 
  So we will install programs in  <code>/home/$USER/Genomics_Workshop</code>, <br>
  while the data and computational output will be held in <code>/scratch/$USER/Genomics_Workshop</code>. </p>
<h2 id="1-install-programs-and-create-a-workspace-for-the-workshop">1. Install programs and create a workspace for the workshop</h2>
<p>Each program has slightly different installation instructions. 
   You do not need to install programs manually.  Instead just run the following scirpt:
   <strong>/projects/oarc/Genomics_Workshop/RNA-Seq_analysis/Labs/lab_PartI.sh</strong><br>
   It will install neccessary programs and creates folders for this workshop.</p>
<p>For curious one, here is the content of the script
   ########################################################
 ```<br />
 #!/bin/bash</p>
<p>mkdir -p /home/$USER/Genomics_Workshop/
mkdir -p /scratch/$USER/Genomics_Workshop/scripts
mkdir -p /scratch/$USER/Genomics_Workshop/download
mkdir -p /scratch/$USER/Genomics_Workshop/untreated
mkdir -p /scratch/$USER/Genomics_Workshop/dex_treated
mkdir -p /scratch/$USER/Genomocs_Workshop/Reference</p>
<p>echo "Copying files... Please wait"</p>
<p>cp  -r /projects/oarc/Genomics_Workshop/Programs/ /home/$USER/Genomics_Workshop/</p>
<p>echo '## Genomics_Workshop specific settings 07/16/2018' &gt;&gt; ~/.bashrc
echo 'export PATH=$HOME/Genomics_Workshop/Programs/seqtk:$PATH' &gt;&gt; ~/.bashrc
echo 'export PATH=$HOME/Genomics_Workshop/Programs/sratoolkit.2.8.2-centos_linux64/bin:$PATH' &gt;&gt; ~/.bashrc
echo 'export PATH=$HOME/Genomics_Workshop/Programs/FastQC:$PATH' &gt;&gt; ~/.bashrc
source ~/.bashrc </p>
<p>module load intel/17.0.2 python/2.7.12
pip install HTSeq --user
wait
pip install bx-python==0.7.3 --user
wait
pip install RSeQC --user</p>
<pre><code>   ########################################################&lt;br&gt;


## 2.  Download data

We will download human RNA-seq data with [GEO accession GSE52778](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52778). The samples  are in NCBI's short read archive format (SRA). 

We will use **sratoolkit** programs to download data but first we need to configure a location where all data files will be stored. &lt;br&gt;
`vdb-config` is a configuration subprogram for `sratoolkit`. We will use it to specify the directory where `sratoolkit` fetches data. You will need to type in the followingt path, but remember to replace `netID` with your own `Rutgers netid`  `/scratch/your_netID/Genomics_Workshop/download`.  Do not copy blindly!
Remember that  **sratoolkit**  is not designed to handle complex downloads. All data will be placed in one folder. You will need to move downloaded files for further analysis/manipulation into different locations manually.
</code></pre>

<pre><code>            vdb-config   --interactive-mode textual     ### dash-dash before interactive-mode
</code></pre>
<pre><code></code></pre>

<h2 id="now-in-the-program">Now in the program:</h2>
<pre><code>                     Your choice &gt; 4
</code></pre>
<h2 id="type-new-path-in">type new path in</h2>
<pre><code>                    /scratch/your_netID/Genomics_Workshop/download
                    Your choice &gt; Y
</code></pre>
<h2 id="hit-enter-and-exit-the-program">Hit Enter and exit the program</h2>
<pre><code>
Then execute the following commands to get the data. Both `prefetch` and `fastq-dump` are part of sratools.&lt;br&gt;
Downloading can be time consuming, it takes about 3 minutes per file. Thus we already downloaded files and placed them in your /scratch/$USER/Genomics_Workshop/download/sra folder. Skim through this section to understand how we did it and continue with 'fast-dump' below.  &lt;br&gt;

</code></pre>

<h1 id="fetch-the-sra-data">fetch the SRA data</h1>
<h1 id="untreated">Untreated</h1>
<p>prefetch -v SRR1039508<br />
prefetch -v SRR1039512<br />
prefetch -v SRR1039516  </p>
<h1 id="dex-treated">DEX treated</h1>
<p>prefetch -v SRR1039509<br />
prefetch -v SRR1039513<br />
prefetch -v SRR1039517                          </p>
<pre><code>Next step is to unpack sra files and convert them to more suitable *fastq* format with  `fastq-dump` program &lt;br&gt;
To unpack the original sequence files can be a bit tricky at first. We need to put them into different directories:&lt;br&gt;
*508,512,516* into /scratch/$USER/Genomics_Workshop/download/untreated and &lt;br&gt; 
 *509,513,517* into /scratch/$USER/Genomics_Workshop/download/dex_treated &lt;br&gt;
 Luckily `fastq-dump` can do processing and output results into specified folders at the same time. 

</code></pre>

<p>cd /scratch/$USER/Genomics_Workshop/download/sra
fastq-dump --outdir /scratch/$USER/Genomics_Workshop/untreated --split-files  SRR10395{08,12,16}.sra
fastq-dump --outdir /scratch/$USER/Genomics_Workshop/dex_treated --split-files  SRR10395{09,13,17}.sra</p>
<pre><code>It takes a while to convert sra files. To save time, files are already converted for you. 
Run the following command to copy files into your /scratch/ directories.
</code></pre>

<p>sra-fastq.sh</p>
<pre><code>
# Running bioinformatics jobs

## 3. FastQC - raw data QC 

FastQC performs a quality control checks on raw sequence data and produces various graphical outputs for visual analysis.
</code></pre>

<pre><code>    cd /scratch/$USER/Genomics_Workshop/untreated         
    module load java  ## fastqc is written in java; we need to load java before using fastqc
    mkdir fastqc      ## create a folder to store the QC output 
    fastqc -o fastqc SRR1039508_1.fastq SRR1039508_2.fastq
</code></pre>
<pre><code>FastQC produces html pages in `fastqc/SRR1039508_1(2)_fastqc.html`, with different kinds of views of data (and Phred scores). You can open this file in Firefox browser. 
</code></pre>

<pre><code>   firefox fastqc/SRR1039508_1_fastqc.html
</code></pre>
<pre><code>To learn more about FastQC, see this pdf file &lt;br&gt;
/projects/oarc/Genomics_Workshop/RNA-Seq_analysis/misc/FastQC_details.pdf &lt;br&gt;
Close Firefox when you are done.

## 4. Trimmomatic - quality trim/adaptor removal

For demonstration purpose, we will take a small subset data using `seqtk` program
</code></pre>

<pre><code>    cd /scratch/$USER/Genomics_Workshop/untreated
    seqtk sample -s100  SRR1039508_1.fastq 10000 &gt; SRR1039508_1_10k.fastq 
    seqtk sample -s100  SRR1039508_2.fastq 10000 &gt; SRR1039508_2_10k.fastq
</code></pre>
<pre><code>More details and examples how to use `seqtk` can be found in &lt;br&gt; 
/projects/oarc/Genomics_Workshop/RNA-Seq_analysis/misc/Seqtk_Examples

Now, run `trimmomatic` to trim the read quality, and remove adaptor&lt;br&gt;

**NOTE:**  trimmomatic command starting with `java -jar` is a one line command, move the slider to the right to see the whole line. 
</code></pre>

<pre><code>    module load java    ### needed for trimmomatic
    java -jar /home/$USER/Genomics_Workshop/Programs/Trimmomatic-0.36/trimmomatic-0.36.jar PE -phred33 -trimlog trim.log SRR1039508_1_10k.fastq SRR1039508_2_10k.fastq SRR1039508_1.paired.fastq SRR1039508_1.unpaired.fastq SRR1039508_2.paired.fastq SRR1039508_2.unpaired.fastq ILLUMINACLIP:/home/$USER/Genomics_Workshop/Programs/Trimmomatic-0.36/adapters/TruSeq3-PE.fa:2:30:10 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:35
</code></pre>
<pre><code>For your convenience we put this command into a bash script, thus you may just type
</code></pre>

<p>run_trimmo.sh</p>
<pre><code>
Once it started run, you should see the following:
</code></pre>

<pre><code>    TrimmomaticPE: Started with arguments:
    -phred33 -trimlog trim.log SRR1039508_1_10k.fastq SRR1039508_2_10k.fastq SRR1039508_1.paired.fastq SRR1039508_1.unpaired.fastq SRR1039508_2.paired.fastq SRR1039508_2.unpaired.fastq ILLUMINACLIP:/home/yc759/Programs/Trimmomatic-0.36/adapters/TruSeq3-PE.fa:2:30:10 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:15 MINLEN:35
    Multiple cores found: Using 2 threads
    Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
    ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
    Input Read Pairs: 100000 Both Surviving: 96596 (96.60%) Forward Only Surviving: 1542 (1.54%) Reverse Only Surviving: 1467 (1.47%) Dropped: 395 (0.40%)
    TrimmomaticPE: Completed surccessfully
</code></pre>
<pre><code>
View the output, the trim.log file, .e.g.  length=63 55 1 56 7 (the original read length 63, now 55 after trim, 1 base from left end and 7 bases from the right end were trimmed off, 56 bases in middle remained)

Alternatively, you may also use `fastx_quality_stats` from *the FASTX—toolkit* ( not covered in this workshop).


## 5. FastQC - Run on cleaned reads
You may run FastQC again on the cleaned by trimmomatic reads and compare new results with results for raw data, step 3 above.
</code></pre>

<pre><code> module load java
 fastqc -o fastqc SRR1039508_1.paired.fastq SRR1039508_2.paired.fastq
</code></pre>
<pre><code>
## 6. Download reference and reference indexing 

Indexing of Human genome may take hours. Thus we have the reference genome prepared and ready for use in &lt;br&gt;
`/projects/oarc/Genomics_Workshop/Reference/ `
&lt;br&gt;
For in class practice we demonstrate how to do genome indexing  with E.coli genome

</code></pre>

<pre><code>    cd /scratch/$USER/Genomics_Workshop/Reference

    wget ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/bacteria/Escherichia_coli/latest_assembly_versions/GCA_000005845.2_ASM584v2/GCA_000005845.2_ASM584v2_genomic.fna.gz

    gunzip GCA_000005845.2_ASM584v2_genomic.fna.gz
    module load bowtie2
    bowtie2-build GCA_000005845.2_ASM584v2_genomic.fna GCA_000005845.2_ASM584v2_genomic
</code></pre>
<pre><code>For your convenience we put these commands into a bash script, thus you may just type
</code></pre>

<p>run_bowtie2.sh</p>
<pre><code>

## 7. Mapping with tophat2, (STAR, HISAT2)
Before we start mapping job, make sure that you WON'T start a job on a login node.
tophat2 jobs are computationally intensive and may require hours to be completed. &lt;br&gt;

- **NOTE: We do not expect you to complete mapping in the class. You may start it to get familiar with the procedure, but do not wait till the completion. You may however do it as a homework for a preparation for the next session ( Genomics Workshop part II).** &lt;br&gt;

Also tophat2 is a multithreaded program, which means it can utilize more than one cpu-core, thus it is better to request more resources with srun command ( so far  we used `srun -c 2` (two cpu-cores).  For tophat2 jobs it is advisable to request more cpu-cores. For the convenience of this workshop we will use `-c 7`, however you may request more cpu-cores for your research.&lt;br&gt;
Exit the current interactive session: type ```exit``` in the terminal wondow. You should see that the prompt in the terminal changed from a compute node to a login node.
`netid@hal0011` to `netid@amarel`
Start a new interactive session requesting more cpu-cores (-c 7) &lt;br&gt;
</code></pre>

<p>srun  -p main --x11 --reservation=genomics -N 1 -c 7 -n 1 -t 01:40:00 --pty /bin/bash -i</p>
<pre><code>For your convenience we put this command into a bash script, thus you may just type
</code></pre>

<p>tophat2_node_request.sh</p>
<pre><code>Notice that the prompt changed again to a compute node, e.g. `netid@hal0011` 

 Go to your folder with data.&lt;br&gt;
 Remember that you should match the `tophat2 -p` option  to be consistent with the number of cores `srun -c` that you requested, thus we use `tophat2 -p 7`, because we requested `srun -c 7`. &lt;br&gt;
**NOTE:**  tophat2 command starting with `tophat2 -p 7` is a one line command, move the slider to the right to see the whole line.        
</code></pre>

<pre><code>    cd  /scratch/$USER/Genomics_Workshop/untreated
    module load mvapich2/2.1  boost/1.59.0  tophat2/2.1.0
    module load samtools  bowtie2 
    mkdir tophat_out
    tophat2 -p 7 --library-type fr-unstranded  -o tophat_out/untreated_SRR1039508_10k --transcriptome-index /projects/oarc/Genomics_Workshop/Reference/hg20_transciptome/GRCh38.78 /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.dna.toplevel SRR1039508_1.paired.fastq SRR1039508_2.paired.fastq
</code></pre>
<pre><code>For your convenience we put these commands into a bash script, thus you may just type
</code></pre>

<p>run_tophat2.sh</p>
<pre><code>
You should see a similar output:
</code></pre>

<h2 id="2018-03-30-114857-beginning-tophat-run-v210">[2018-03-30 11:48:57] Beginning TopHat run (v2.1.0)</h2>
<p>[2018-03-30 11:48:57] Checking for Bowtie
                  Bowtie version:        2.2.9.0
[2018-03-30 11:48:58] Checking for Bowtie index files (transcriptome)..
[2018-03-30 11:48:58] Checking for Bowtie index files (genome)..
[2018-03-30 11:48:58] Checking for reference FASTA file
[2018-03-30 11:48:58] Generating SAM header for /projects/oarc/Genomics_Workshop/Referen
ce/hg20/Homo_sapiens.GRCh38.dna.toplevel
[2018-03-30 11:49:09] Reading known junctions from GTF file
[2018-03-30 11:49:27] Preparing reads
         left reads: min. length=35, max. length=63, 96592 kept reads (4 discarded)
        right reads: min. length=35, max. length=63, 96594 kept reads (2 discarded)
[2018-03-30 11:49:29] Using pre-built transcriptome data..
[2018-03-30 11:49:35] Mapping left_kept_reads to transcriptome GRCh38.78 with Bowtie2
[2018-03-30 11:49:49] Mapping right_kept_reads to transcriptome GRCh38.78 with Bowtie2
[2018-03-30 11:50:03] Resuming TopHat pipeline with unmapped reads
[2018-03-30 11:50:03] Mapping left_kept_reads.m2g_um to genome Homo_sapiens.GRCh38.dna.t
oplevel with Bowtie2
[2018-03-30 11:50:16] Mapping left_kept_reads.m2g_um_seg1 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (1/2)
[2018-03-30 11:50:18] Mapping left_kept_reads.m2g_um_seg2 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (2/2)
[2018-03-30 11:50:20] Mapping right_kept_reads.m2g_um to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2
[2018-03-30 11:50:23] Mapping right_kept_reads.m2g_um_seg1 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (1/2)
[2018-03-30 11:50:25] Mapping right_kept_reads.m2g_um_seg2 to genome Homo_sapiens.GRCh38.dna.toplevel with Bowtie2 (2/2)
……………………………………………………….</p>
<pre><code>Here is another example of mapping with tophat. &lt;br&gt;
The transcriptome index was built  by pointing to a gtf file first (the gtf file is prepared earlier).  The following would be the command to generate the transcriptome index while running tophat alignment.&lt;br&gt;
**NOTE:**  tophat2 command starting with `tophat2 -p 7` is a one line command, move the slider to the right to see the whole line. 
</code></pre>

<p>tophat2 -p 7 --library-type fr-unstranded  -o tophat_out/untreated_SRR1039508 –GTF /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.78.gtf --transcriptome-index /projects/oarc/Genomics_Workshop/Reference/hg20_transciptome/GRCh38.78 /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.dna.toplevel SRR1039508_1.fastq.pairedOut.fastq SRR1039508_2.fastq.pairedOut.fastq</p>
<pre><code>After the completion the output folder `tophat_out/untreated_SRR1039508/` shall contain the following files/folders:  
</code></pre>

<p>-rw-rw-r-- 1 yc759 oarc 2174796848 Jan 16 21:57 accepted_hits.bam
 -rw-rw-r-- 1 yc759 oarc        565 Jan 16 21:57 align_summary.txt
 -rw-rw-r-- 1 yc759 oarc    1921529 Jan 16 21:57 deletions.bed
 -rw-rw-r-- 1 yc759 oarc    2239884 Jan 16 21:57 insertions.bed
 -rw-rw-r-- 1 yc759 oarc   14181618 Jan 16 21:57 junctions.bed
 drwxrwsr-x 2 yc759 oarc       4096 Jan 16 21:57 logs
 -rw-rw-r-- 1 yc759 oarc        184 Jan 16 21:57 prep_reads.info
 -rw-rw-r-- 1 yc759 oarc   42846571 Jan 16 21:57 unmapped.bam</p>
<pre><code>
## 8. Read counts using htseq-count

Tophat mapping is resource intense and time consuming. If you managed to produce the bam file yourself, it's great. If not, let's move forward and copy files that we prepared for you.

</code></pre>

<pre><code> cd /scratch/$USER/Genomics_Workshop/untreated/tophat_out/untreated_SRR1039508 
 cp /projects/oarc/Genomics_Workshop/RNA-Seq_analysis/SRA_data/untreated/tophat_out/untreated_SRR1039508/accepted_hits.bam accepted_hits.bam
</code></pre>
<pre><code>
Next, we will use samtools to sort the bam file by name:  because htseq-count accepts bam file sorted by **name** as default, but tophat generates bam sorted by **coordinates** as default. 
</code></pre>

<pre><code> module load samtools intel/17.0.2 python/2.7.12   #loads the software
 samtools sort -n  accepted_hits.bam | samtools view | htseq-count -m intersection-nonempty -t exon -i gene_id -s no --additional-attr=gene_name - /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.78.gtf &gt; untreated08.txt
</code></pre>
<pre><code>In the same way, generate the counts file `untreated12.txt`, `untreated16.txt`,`dex09.txt`, `dex13.txt`, `dex17.txt`. &lt;br&gt;
To save time and for your convenience we prepared files already, thus you may just type
</code></pre>

<p>run_htseq_count.sh</p>
<pre><code>
## 9. Perform Mapping QC using RSeQC

Now,  quality control using RSeQC - a few examples are given here, please go to the [rseqc website](http://rseqc.sourceforge.net/) for more functions.&lt;br&gt; 
- Example 1 &lt;br&gt;
Execute:   
</code></pre>

<p>cd /scratch/$USER/Genomics_Workshop/untreated/tophat_out/untreated_SRR1039508 
module load python/2.7.12
module load intel/17.0.4</p>
<p>read_distribution.py -i accepted_hits.bam -r /projects/oarc/Genomics_Workshop/Reference/Homo_sapiens.GRCh38.79.bed</p>
<pre><code>Provided a BAM/SAM file and reference gene model, `read_distribution.py` will calculate how mapped reads were distributed over genome feature (like CDS exon, 5’UTR exon, 3’ UTR exon, Intron, Intergenic regions).&lt;br&gt;
Output: 
</code></pre>

<pre><code>    processing/projects/oarc/Genomics_Workshop/Reference/Homo_sapiens.GRCh38.79.bed ... Done
    processing accepted_hits.bam ... Finished

    Total Reads                   43474036
    Total Tags                    54438789
    Total Assigned Tags           53991382
    =====================================================================
    Group               Total_bases         Tag_count           Tags/Kb
    CDS_Exons           103371993           43264842            418.54
    5'UTR_Exons         5217678             583447              111.82
    3'UTR_Exons         29324747            8145122             277.76
    Introns             1500197093          1805034             1.20
    TSS_up_1kb          33306654            18893               0.57
    TSS_up_5kb          148463534           41165               0.28
    TSS_up_10kb         265823549           55644               0.21
    TES_down_1kb        35215293            50954               1.45
    TES_down_5kb        152556214           113325              0.74
    TES_down_10kb       268614580           137293              0.51
    =====================================================================
</code></pre>
<pre><code>- Example 2 &lt;br&gt;
Execute:  
</code></pre>

<p>bam_stat.py -i accepted_hits.bam</p>
<pre><code>It will summarize mapping statistics of a BAM or SAM file.&lt;br&gt;
Output: 
</code></pre>

<pre><code>    Load BAM file ...

    Done

    #==================================================
    #All numbers are READ count
    #==================================================

    Total records:                          52528699

    QC failed:                              0
    Optical/PCR duplicate:                  0
    Non primary hits                        9054663
    Unmapped reads:                         0
    mapq &lt; mapq_cut (non-unique):           2684801

    mapq &gt;= mapq_cut (unique):              40789235
    Read-1:                                 20414530
    Read-2:                                 20374705
    Reads map to '+':                       20393901
    Reads map to '-':                       20395334
    Non-splice reads:                       30860931
    Splice reads:                           9928304
    Reads mapped in proper pairs:           32386536
    Proper-paired reads map to different chrom:312
</code></pre>
<pre><code>- Example 3 &lt;br&gt;
In this  example we calculate the RNA-seq reads coverage over gene body using `geneBody_coverage.py`.
However, this script requires  that bam files to be sorted and indexed. We will do it using `samtools`. Moreover for calculations and generation of  graphical outputs  R-package is needed.&lt;br&gt;
Load `samtools` and `R-Project` modules and go to the tophat_out sample folder

</code></pre>

<p>module load intel/17.0.4  R-Project/3.4.1  <br />
module load samtools
samtools sort accepted_hits.bam -o accepted_hits.sorted.bam </p>
<pre><code>This may take some time, thus you may use instead one of already prepared files by making a soft link to it:
</code></pre>

<p>ln -s /projects/oarc/Genomics_Workshop/RNA-Seq_analysis/SRA_data/untreated/tophat_out/untreated_SRR1039508/accepted_hits.sorted.bam accepted_hits.sorted.bam</p>
<pre><code>Execute:
</code></pre>

<p>samtools index accepted_hits.sorted.bam
geneBody_coverage.py -r /projects/oarc/Genomics_Workshop/Reference/hg38.housekeepingGenes.bed -i accepted_hits.sorted.bam -o test</p>
<pre><code>Output:
</code></pre>

<p>@ 2018-01-14 13:17:33: Read BED file (reference gene model) ...
@ 2018-01-14 13:17:33: Total 3802 transcripts loaded
@ 2018-01-14 13:17:33: Get BAM file(s) ...
        accepted_hits.sorted.bam
@ 2018-01-14 13:17:33: Processing accepted_hits.sorted.bam ...
        3800 transcripts finished</p>
<pre><code>    Sample  Skewness
    accepted_hits.sorted    -3.61577607436
</code></pre>
<p>@ 2018-01-14 13:28:59: Running R script ...
null device
          1</p>
<pre><code>output files:&lt;br&gt;
test.geneBodyCoverage.r &lt;br&gt;
test.geneBodyCoverage.txt &lt;br&gt;
log.txt &lt;br&gt;
test.geneBodyCoverage.curves.pdf &lt;br&gt;

- Example 4 &lt;br&gt;
Checking  whether the sequencing read is strand specific or not &lt;br&gt;
Download the gene.bed files :  `https://sourceforge.net/projects/rseqc/files/BED/Human_Homo_sapiens/`  
**NOTE** Be careful, the genome version has to  be consistent with the  reference genome used in mapping.   &lt;br&gt;

Execute:
</code></pre>

<p>infer_experiment.py -r  /projects/oarc/Genomics_Workshop/Reference/Homo_sapiens.GRCh38.79.bed -i accepted_hits.bam</p>
<pre><code>Output:
</code></pre>

<p>Reading reference gene model /projects/oarc/Genomics_Workshop/Reference/Homo_sapiens.GRCh38.79.bed ... Done
Loading SAM/BAM file ...  Total 200000 usable reads were sampled</p>
<p>This is PairEnd Data
Fraction of reads failed to determine: 0.1406
Fraction of reads explained by "1++,1--,2+-,2-+": 0.4302
Fraction of reads explained by "1+-,1-+,2++,2--": 0.4292</p>
<pre><code>
## 10. R practice 
There is one exeption when the computational job can be started on the cluster login node,  and it is when you need to install some software packages. `R program` is alread installed on the cluster as a module, but we need to add some packages from [bioconductor](https://bioconductor.org) as well. &lt;br&gt;
- a). Install needed packages
exit from the compute node 
    ```
    exit
    ```  
You should see that the prompt in the terminal changed from a compute node to a login node.
`netid@hal0011` to `netid@amarel`
Now load R module on a login node

</code></pre>

<p>module load intel/17.0.4
module load R-Project/3.4.1
R</p>
<pre><code>In R environmnent install the following packages
</code></pre>

<pre><code>source("https://bioconductor.org/biocLite.R") 
biocLite("ape")
biocLite("MKmisc")
biocLite("Heatplus")
biocLite("affycoretools")
biocLite("flashClust")
biocLite("affy")
##yes (for personal library)
</code></pre>
<pre><code>Exit R:
</code></pre>

<pre><code>&gt; quit()
</code></pre>
<p>Save workspace image? [y/n/c]: n</p>
<pre><code>- b). Download gene annotation in gtf format from [Ensembl](http://useast.ensembl.org/index.html)&lt;br&gt;
</code></pre>

<p>/scratch/$USER/Genomics_Workshop/Reference/
wget ftp://ftp.ensembl.org/pub/release-78/gtf/homo_sapiens/Homo_sapiens.GRCh38.78.gtf.gz
tar -zxvf Homo_sapiens.GRCh38.78.gtf.gz</p>
<pre><code>*NOTE*: the latest Ensembl Release is 93 (July 2018) but we will work with release 78 for this class
To save time you may just copy gtf file:
</code></pre>

<p>cp /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.78.gtf /scratch/$USER/Genomics_Workshop/Reference/
```</p>
<p>Calculate a gene length  </p>
<pre><code>###cp  the Homo_sapiens.GRCh38.78.gtf to your own Reference folder  
cp /projects/oarc/Genomics_Workshop/Reference/hg20/Homo_sapiens.GRCh38.78.gtf /scratch/$USER/Genomics_Workshop/Reference/.



module load intel/17.0.4
module load R-Project/3.4.1
##start R on computer node now
srun --x11 -p main --reservation=ddclass -N 1 -c 2 -n 1 -t 01:40:00 --pty /bin/bash


&gt; library(GenomicFeatures)
&gt; gtfdb &lt;- makeTxDbFromGFF("Homo_sapiens.GRCh38.78.gtf",format="gtf")
&gt; exons.list.per.gene &lt;- exonsBy(gtfdb,by="gene")
&gt; exonic.gene.sizes &lt;- lapply(exons.list.per.gene,function(x){sum(width(reduce(x)))})
&gt; class(exonic.gene.sizes)

&gt; Hg20_geneLength &lt;-do.call(rbind, exonic.gene.sizes)
&gt; colnames(Hg20_geneLength) &lt;- paste('geneLength')  
&gt; write.csv(Hg20_geneLength, file="Hg20_geneLength") ##export/save this gene length file into your working directory
### the gene length is defined to be the total length of all exons in the gene, including the 3'UTR###

c). More self practice

           ##at R prompt


     &gt; options(stringsAsFactors = FALSE)   #this will prevent R from reading in character data as factor data


#### Arithmetic functions

2+2
3*3
3*8+2
log10(1000)
log2(8)
abs(-10)
sqrt(81)


#### Creating objects

ls()  #see what objects are in the workspace
x &lt;- 4
x
x = 3  #a single = is an assignment operator
x
x == 5 #a double == asks "is the left side equivalent to the right side?"
x + 2   #objects can be used in equations
y &lt;- "anyname"
y
class(x)
class(y)
ls()


#### Vector and Matrix

x1 &lt;- c(1,2,3,4,5)
x1
class(x1)
length(x1)
x &lt;- cbind(x1, x1+1)    #1 will be added to all the numbers in x1
x
class(x)       #what kind of object is x?
dim(x)         #the dimension of matrix
x1[1:3]        #use [] to get subsets of a vector
x[1,]          #use [,] to get subsets of a matrix (or dataframe)
x[,1]
x[,-1]
x[c(1,2),]
x[-c(1,3),]
colnames(x)
colnames(x) &lt;-c("A","B")
rownames(x) &lt;-c("C","D","E","F","G")
x

#### Data Frames

z &lt;- data.frame(A=x[,1], B=rownames(x), C=factor(rownames(x)), D=x[,1]==3, stringsAsFactors=F)
class(z)
names(z)
dim(z)
class(z$A)
class(z$B)
class(z$C)
class(z$D)
z$B
z$C


#### More ways to subset dataframes

z$B
z[[2]]
z[,2]   #these first 3 give equivalent results
z[,1:2]
z[,c(1,3)]
z[c(1,3:5),]


#### Lists

mylist &lt;- list(first=z,second=x,third=c("W","X","Y","Z"))
class(mylist)
mylist
names(mylist)
class(mylist$first)
class(mylist$second)


#### Functions

my.add &lt;- function(a, b) {a - b}
class(my.add)
my.add(4,99)
my.add(99,4)
my.add(b = 99, a = 4)


library(limma)  #load the limma package


#### Make sure the working directory is set to your file on the computer;

getwd()  #see what the current working directory is
setwd("????????????????")  #change the working directory


#### Output a single object as a comma separated value file

write.csv(z, file="test.csv")



#### Save all the objects you have created to your workspace

save.image()                #creates a default file named ".RData"
save.image("intro.Rdata")   #creates a named file


#### Remove objects from your workspace

ls()
rm(x)          #remove a single object by name
ls()
rm(z,x1)       #remove multiple objects by name
ls()
load("intro.Rdata")
ls()
rm(list=ls())  #remove all objects
ls()


#### Save a history of all the commands entered

savehistory("introhistory.Rhistory")
</code></pre>
              
            </div>
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/rutgers-oarc/training/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
      
    </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme.js" defer></script>
      <script src="../../search/main.js" defer></script>

</body>
</html>
